src/Microsoft.OData.Core/ODataResourceTypeContext.cs

```diff
@@ -55,23 +55,23 @@ private ODataResourceTypeContext(IEdmStructuredType expectedResourceType, bool t
         /// </summary>
         public virtual string NavigationSourceName
         {
-            get { return this.ValidateAndReturn(default(string)); }

```

Comment:
can we just use default property get?

public virtual string NavigationSourceName { get; }

Reply:
:thumbsup: 

yes; that would be simpler. good catch.